bench.py

```diff
@@ -119,15 +119,35 @@ def calc_latency_stats(queries, duration, min_latency, max_latency,
     return data
 
 
-def process_results(results):
-    try:
-        lat_data = json.loads(results)
-    except json.JSONDecodeError as e:
-        print('could not process benchmark results: {}'.format(e),
-              file=sys.stderr)
-        print(results, file=sys.stderr)
-        sys.exit(1)
-
+def aggregate_results(raw_data):
+    for queries_bench in raw_data['data']:
+        # effectively the duration is multiplied by the number of
+        # queries, since we aggregate results over all of them
+        queries_bench['duration'] *= len(queries_bench['queries'])
```

Comment:
We should use geometric mean here instead of summing/maxing things: https://github.com/MagicStack/pgbench/blob/cfd72c983c4d779640b005221d0885917d52bf7b/pgbench#L120-L151